test/factorials.jl

```diff
@@ -11,6 +11,7 @@
     # derangement
     @test derangement(4) == subfactorial(4) == 9
     @test derangement(24) == parse(BigInt, "228250211305338670494289")
```

Comment:
Add more test to cover branch: `elseif n <= 1`
```suggestion
    @test derangement(0) == 1
    @test derangement(1) == 0
    @test derangement(24) == parse(BigInt, "228250211305338670494289")
```